compiler/modules/front/parseExprs.bal

```diff
@@ -0,0 +1,170 @@
+import wso2/nballerina.err;
+
+type Identifier readonly & record {|
+    string identifier;
+    err:Position pos;
+|};
+
+type LiteralType DECIMAL_NUMBER|STRING_LITERAL|BOOLEAN_LITERAL;
+
+type LHSExprNode Expr|Identifier;
```

Comment:
This LHS/RHS distinction seems very strange to me. I don't know what this is achieving